### PR TITLE
Faster metrics calculations + Fix warnings added by the new version of torchmetrics

### PR DESCRIPTION
**Pull request by dskhudia**
# What does this PR do?

Newer version of torchmetrics can be faster if we update our custom metrics.

# What issue(s) does this change relate to?

Composer generating extra warning

TEST:

No warning with this change. 
```
import torch
from torchmetrics.utilities import check_forward_full_state_property



from composer.metrics import MaskedAccuracy

batch_size = 16
num_classes = 10
generated_preds = torch.rand((batch_size, num_classes))
true_labels = torch.randint(low=0, high=num_classes - 1, size=(batch_size,))

m = MaskedAccuracy(ignore_index=0)
m.update(generated_preds.float(), true_labels.float())
```

```
import torch
from torchmetrics.utilities import check_forward_full_state_property

from composer.metrics import MaskedAccuracy

batch_size = 16
num_classes = 10
generated_preds = torch.rand((batch_size, num_classes))
true_labels = torch.randint(low=0, high=num_classes - 1, size=(batch_size,))

check_forward_full_state_property(
    MaskedAccuracy,
    init_args={"ignore_index": 0},
    input_args={"preds": generated_preds.float(), "target": true_labels.float()},
)
```

### CODE REVIEW

**Review by mvpatel2000 — COMMENTED** [review]
Can we remove link to blog post? I think it's probably fine to just put the comment. Also adding vision people for review for metrics. I assume you have the NLP ones covered

**Review by Landanjs — APPROVED** [review]
LGTM

I tested mIoU and Accuracy for ADE20k and ImageNet, respectively. Early values seem to be fine, but it will take sometime to get convergence numbers. Throughput did not seem to change.

**Comment by dskhudia** [comment]
@mvpatel2000 Removed the link.

**Review by mvpatel2000 — APPROVED** [review]
